### ISSUE TITLE
Support priority when running in parallel

### ISSUE THREAD

**Issue by a-vishar** [microsoft/testfx#572]
When using the parallel test running feature, it would be helpful to specify priority of some kind.

The issue we have is that we have a number of longer running tests (< core count), and a large number of quick running tests.  The longer running tests each take approximately the same time as the sum time of the shorter ones (7 hours).

If we could prioritise the long running ones and leave one core free for the quick ones, our entire run would complete in 7 hours.

What currently happens is that some of the long running tests will start immediately, the rest of the cores will be used for some number of hours by the short running tests, then the last long running test starts multiple hours in, leading to a total run-time of 11 hours.

To illustrate it more clearly:

- Imagine having only two cores
- One test which lasts an hour
- 1000 tests which (when run on one core) last an hour

An optimal scheduler would lead to a run-time of 1 hour (long test on one core, short tests on the other).

A sub-optimal scheduler would run run the quick tests on both cores (30 minutes) then run the long running test (1 hour), leading to a total run-time of 90 minutes.

As I have knowledge of how long the tests take, designing the scheduler manually is easy!

Support for an attribute like `[TestPriority(int.MaxValue)]` (default zero) would allow be to set the priority very easily in my existing test suite with minimal effort.

**Comment by jayaranigarg** [comment]
@a-vishar : Thank you for your feedback. 

This is a new ask and seems related to running the test methods in some particular order. We are going to mark this an enhancement currently and will love to pick it up in the future, if we have enough votes on it.

**Comment by Evangelink** [comment]
I'll close as won't fix given the few upvotes or request for such feature.
